amr-wind/equation_systems/AdvOp_Godunov.H

```diff
@@ -123,17 +133,52 @@ struct AdvectionOp<
                         });
                 }
 
-                amrex::FArrayBox tmpfab(amrex::grow(bx, 1), PDE::ndim * 14);
-
-                godunov::compute_fluxes(
-                    lev, bx, PDE::ndim, (*flux_x)(lev).array(mfi),
-                    (*flux_y)(lev).array(mfi), (*flux_z)(lev).array(mfi),
-                    (PDE::multiply_rho ? rhotrac : tra_arr),
-                    u_mac(lev).const_array(mfi), v_mac(lev).const_array(mfi),
-                    w_mac(lev).const_array(mfi), src_term(lev).const_array(mfi),
-                    dof_field.bcrec_device().data(), iconserv.data(),
-                    tmpfab.dataPtr(), geom, dt, godunov_scheme);
-
+                if ((godunov_scheme == godunov::scheme::PPM_NOLIM) ||
+                    (godunov_scheme == godunov::scheme::WENOJS) ||
+                    (godunov_scheme == godunov::scheme::WENOZ)) {
+                    amrex::FArrayBox tmpfab(amrex::grow(bx, 1), PDE::ndim * 14);
+
+                    godunov::compute_fluxes(
+                        lev, bx, PDE::ndim, (*flux_x)(lev).array(mfi),
+                        (*flux_y)(lev).array(mfi), (*flux_z)(lev).array(mfi),
+                        (PDE::multiply_rho ? rhotrac : tra_arr),
+                        u_mac(lev).const_array(mfi),
+                        v_mac(lev).const_array(mfi),
+                        w_mac(lev).const_array(mfi),
+                        src_term(lev).const_array(mfi),
+                        dof_field.bcrec_device().data(), iconserv.data(),
+                        tmpfab.dataPtr(), geom, dt, godunov_scheme);
+                } else {
```

Comment:
I presume this is for BDS? Can it be an `else if` to make it clear?

Reply:
It's not! lol it's for PPM, PLM, _and_ BDS. So we don't have to roll our own code! it's super sweet. Unfortunately PPM_NOLIM, and WENO variants are not in amrex-hydro so I can't blow away as much could as I would have liked ;)

Reply:
Is it still possible to make it an `else if`?

Reply:
yup, done